src/scancode/resource.py

```diff
@@ -1497,26 +1497,75 @@ def get_resource_basic_attributes(_resource_data):
             _is_file = _file_type == 'file'
             return _name, _path, _is_file
 
-        # Create root resource
-        root_name, root_path, root_is_file = get_resource_basic_attributes(root_data)
+        def create_empty_resource_data():
+            # Get fields from the base Resource class and the ScannedResource class
+            base_fields = attr.fields(Resource)
+            resource_fields = attr.fields(self.resource_class)
+            # Create dict of {field: field_default_value} for the dynamically created fields
+            resource_data = {}
+            for field in resource_fields:
+                if field in base_fields:
+                    # We only want the fields that are not part of the base set of fields
+                    continue
+                value = field.default
+                if isinstance(value, attr.Factory):
+                    # For fields that have Factories as values, we set their values to be an
+                    # instance of whatever type the factory makes
+                    value = value.factory()
+                resource_data[field.name] = value
+            return resource_data
+
+        # Create root resource without setting root data just yet. If we run into the root data
+        # while we iterate through `resources_data`, we fill in the data then.
+        sample_resource_path = sample_resource_data.get('path')
+        root_path = sample_resource_path.split('/')[0]
```

Comment:
A naive split does not work in all cases. You should strip the paths from slashes first as a precaution
```
>>> '/this/is/'.split('/')[0]
''
```

This would happen if we used `--full-root` option

Reply:
I didn't know about this option. I'll make the change.

Reply:
that works now. An alternative could have bee to use the commoncode.fileutils/paths utility functions too